# HG changeset patch
# User Rik <rik@octave.org>
# Date 1357186708 28800
#      Wed Jan 02 20:18:28 2013 -0800
# Node ID 84b38d6442e685ebc1f57f309f3c43f0c8c55447
# Parent  c18692212b90a0d8a4b1e7daf2d2639d0e87600a
configure.ac: Replace uses of 'true' as a command.

* configure.ac: Replace uses of 'true' as a command.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -1415,19 +1415,17 @@ AC_ARG_ENABLE([dl],
     [disable loading of dynamically linked modules])],
   [case $enableval in
      yes) ENABLE_DYNAMIC_LINKING=true ;;
      no) ENABLE_DYNAMIC_LINKING=false ;;
      *) AC_MSG_ERROR([bad value $enableval for --enable-dl]) ;;
    esac],
   [ENABLE_DYNAMIC_LINKING=true])
 
-if $STATIC_LIBS || $SHARED_LIBS; then
-  true
-else
+if ! $STATIC_LIBS && ! $SHARED_LIBS; then
   AC_MSG_ERROR([You can't disable building static AND shared libraries!])
 fi
 
 CPICFLAG=-fPIC
 CXXPICFLAG=-fPIC
 FPICFLAG=-fPIC
 SHLEXT=so
 SHLLIB='$(SHLEXT)'
@@ -1781,17 +1779,17 @@ if $SHARED_LIBS || $ENABLE_DYNAMIC_LINKI
     ;;
   esac
 
   DL_LIBS="$lt_cv_dlopen_libs"
   AC_SUBST(DL_LIBS)
 
   ## Disable dynamic linking if capability is not present.
   if $dlopen_api || $shl_load_api || $loadlibrary_api || $dyld_api; then
-    true
+    :  # some form of dynamic linking present
   else
     ENABLE_DYNAMIC_LINKING=false
   fi
 fi
 
 if $ENABLE_DYNAMIC_LINKING; then
   AC_DEFINE(ENABLE_DYNAMIC_LINKING, 1, [Define to 1 if using dynamic linking.])
 fi
